pkg/capacity/manager_test.go

```diff
@@ -8,323 +8,171 @@ import (
+	t.Run("with one cluster and enough capacity", func(t *testing.T) {
 		// given
-		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxNumberOfSpaces(1), spc.MaxMemoryUtilizationPercent(60))
-		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxNumberOfSpaces(1), spc.MaxMemoryUtilizationPercent(95))
-		fakeClient := commontest.NewFakeClient(t, toolchainStatus, spc1, spc2)
-		InitializeCounters(t, toolchainStatus)
+		spaceProvisionerConfig := hspc.NewEnabledValidTenantSPC("member1")
+		status := test.NewToolchainStatus()
+		fakeClient := commontest.NewFakeClient(t, spaceProvisionerConfig, status)
+		test.InitializeCounters(t, status)
+		cm := capacity.NewClusterManager(commontest.HostOperatorNs, runtimeclient.Client(fakeClient))
 
 		// when
-		clusterName, err := capacity.NewClusterManager(commontest.HostOperatorNs, fakeClient).GetOptimalTargetCluster(
-			ctx,
-			capacity.OptimalTargetClusterFilter{
-				PreferredCluster:         "member2",
-				ToolchainStatusNamespace: commontest.HostOperatorNs,
-			},
-		)
+		clusterName, err := cm.GetOptimalTargetCluster(ctx, capacity.OptimalTargetClusterFilter{})
 
 		// then
 		require.NoError(t, err)
-		assert.Equal(t, "", clusterName)
+		assert.Equal(t, "member1", clusterName)
 	})
 
-	t.Run("with two clusters but only the second one has enough capacity - using the default values", func(t *testing.T) {
+	t.Run("with three eligible clusters it returns the one with the most capacity (the first one)", func(t *testing.T) {
 		// given
-		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxMemoryUtilizationPercent(62))
-		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxMemoryUtilizationPercent(62))
-		fakeClient := commontest.NewFakeClient(t, toolchainStatus, spc1, spc2)
-		InitializeCounters(t, toolchainStatus)
-
-		// when
-		clusterName, err := capacity.NewClusterManager(commontest.HostOperatorNs, fakeClient).GetOptimalTargetCluster(
-			ctx,
-			capacity.OptimalTargetClusterFilter{
-				ToolchainStatusNamespace: commontest.HostOperatorNs,
-			},
-		)
+		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxNumberOfSpaces(10000), spc.WithConsumedSpaceCount(700))
+		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxNumberOfSpaces(1000), spc.WithConsumedSpaceCount(700))
+		spc3 := hspc.NewEnabledValidTenantSPC("member3", spc.MaxNumberOfSpaces(1000), spc.WithConsumedSpaceCount(200))
+		status := test.NewToolchainStatus()
 
-		// then
-		require.NoError(t, err)
-		assert.Equal(t, "member2", clusterName)
-	})
-
-	t.Run("with two clusters but none of them has enough capacity - using the default memory values", func(t *testing.T) {
-		// given
-		spc1 := hspc.NewEnabledValidTenantSPC("member1", spc.MaxMemoryUtilizationPercent(1))
-		spc2 := hspc.NewEnabledValidTenantSPC("member2", spc.MaxMemoryUtilizationPercent(1))
-		fakeClient := commontest.NewFakeClient(t, toolchainStatus, spc1, spc2)
-		InitializeCounters(t, toolchainStatus)
+		test.InitializeCounters(t, status)
```

Comment:
this doesn't actually test the space counter, does it? The ToolchainStatus is empty, which means that the counter will be empty as well
consider using the helper functions to simplify the initialization of the that I mentioned in the previous review:  https://github.com/codeready-toolchain/host-operator/pull/1119#discussion_r1910444037